### PR TITLE
Fix where expanding resizer wouldn't expanse grid view  

### PR DESCRIPTION
**Pull request by Adaverse**
Below is the issue - 

https://github.com/apache/airflow/assets/35839624/f7c7a30d-c1c5-485b-b171-715baa979e4c

Fix contains the following - 

-  It expands the Grid view when user moves the resize cursor to the right to a minimum width.
-  The grid is expanded when user makes the movement to the right and does not wait until the user crosses the minimum width mark since it makes UI look unresponsive. 

https://github.com/apache/airflow/assets/35839624/81c1c26a-13e9-41c4-9039-683a0cf0b539

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/www/static/js/dag/Main.tsx [hunk 1]
```diff
@@ -108,15 +110,21 @@ const Main = () => {
...
-        const width = `${e.x}px`;
-        gridEl.style.width = width;
-        saveWidth(width);
+      if (gridEl) {
+        if (e.x > minPanelWidth && e.x < window.innerWidth - minPanelWidth) {
+          const width = `${e.x}px`;
+          gridEl.style.width = width;
+          saveWidth(width);
+        } else if (
+          e.x < minPanelWidth &&
```
Comment (bbovenzi) [inline airflow/www/static/js/dag/Main.tsx:132]: What if `e.x === minPanelWidth`?

Comment (Adaverse) [reply airflow/www/static/js/dag/Main.tsx:132]: So the logic states that if we are beyond `minPanelWidth` then adjust the resizer based on mouse movement and when we are lesser than `minPanelWidth` then given the mouse is moving right, snap to the `minPanelWidth` position. 

TL;DR --> Here, we care about what the mouse is doing when not at `minPanelWidth`. But the movement is tracked whether regardless of mouse being at `minPanelWidth` or not.

Comment (Adaverse) [reply airflow/www/static/js/dag/Main.tsx:132]: Should I also add the other way around i.e when a user tries to reduce the size beyond `minPanelWidth` then it should automatically collapse?

Comment (bbovenzi) [reply airflow/www/static/js/dag/Main.tsx:132]: Not a bad idea! Its certainly weird to have set width states and a variable draggable column, but if we can integrate them well then that would be best.

Comment (Adaverse) [reply airflow/www/static/js/dag/Main.tsx:132]: Let's integrate it and see the UI experience. We need to have some minima anyways.

Comment (Adaverse) [reply airflow/www/static/js/dag/Main.tsx:132]: Implemented the other way around as well. But have kept a heuristic where the user needs to cross `minPanelWidth/2` in order to collapse, since there would be cases where the user would reduce the size without intent to collapse. The intent here would be to cross the halfway mark. Let me know your opinion on this.

Have also added an updated demo!

Comment (bbovenzi) [reply airflow/www/static/js/dag/Main.tsx:132]: Looks good! Mind to leave some comments for all this if/else logic? Then I think its good to merge

Comment (Adaverse) [reply airflow/www/static/js/dag/Main.tsx:132]: Added!

**Review by bbovenzi — APPROVED** [review]

**Review by pierrejeambrun — APPROVED** [review]
